modules/swagger-codegen/src/main/resources/csharp/ApiClient.mustache

```diff
@@ -165,24 +165,12 @@ namespace {{packageName}}.Client
 
             if (postBody != null) // http body (model or byte[]) parameter
             {
-                if (postBody.GetType() == typeof(String))
-                {
-                    {{#netStandard}}
-                    request.AddParameter(new Parameter { Value = postBody, Type = ParameterType.RequestBody, ContentType = "application/json" });
-                    {{/netStandard}}
-                    {{^netStandard}}
-                    request.AddParameter("application/json", postBody, ParameterType.RequestBody);
-                    {{/netStandard}}
-                }
-                else if (postBody.GetType() == typeof(byte[]))
```

Comment:
This check previously existed to support file inputs. Can you verify your change works when posting files?

Reply:
@jimschubert I implemented the 'UploadFile' test for the generated Petstore csharp api solution. The test passed and the api response was a 200.  Let me know if there is any further testing you would like me to do.

Reply:
@alex-fisher I don't see that in this PR. Is it on another branch?

Reply:
This was done in the tests generated by the sdk and not in the actual testing suite.  Would you like me to add a permanent test for this?

Reply:
Is it repeatable? If you share it here as a snippet, we can evaluate and either add it to this pr or to a later pr. It would be ideal as a permanent test.

Reply:
This is similar to the test I wrote to verify the upload functionality.  In my case I used an actual jpeg on my file system but for testing purposes a 'fake' file memory stream may suffice. 
```
[Test()]
public void TestFileUpload()
{
	PetApi petApi = new PetApi();
	long petId = 12312;

	using (var testStream = new MemoryStream(Encoding.UTF8.GetBytes("Test File Contents")))
	{
		var response = petApi.UploadFileWithHttpInfo(petId, null, testStream);
		Assert.AreEqual(response.StatusCode, (int)HttpStatusCode.OK);
		Assert.IsTrue(
			response.Data.Message.Contains("File uploaded to ./no_file_name_provided, 18 bytes"));
	}
}
``` 

After looking into this further it may make sense to test the ApiClient directly in ApiClientTests.cs:
```
[Test()]
public void TestFileUpload()
{
	ApiClient client = new ApiClient();
	long petId = 12312;

	var path = "/pet/{petId}/uploadImage";
	var queryParams = new List<KeyValuePair<string, string>>();
	object body = null;
	var headers = new Dictionary<string, string>() {{"Accept", "application/json"}};
	var formParams = new Dictionary<string, string>();
	var pathParams = new Dictionary<string, string>() {{"petId", petId.ToString()}};
	var contentType = "multipart/form-data";

	using (var testStream = new MemoryStream(Encoding.UTF8.GetBytes("Test File Contents")))
	{
		var response = (IRestResponse) client.CallApi(
			path,
			Method.POST,
			queryParams,
			body,
			headers,
			formParams,
			new Dictionary<string, FileParameter>()
			{
				{
					"file",
					client.ParameterToFile("file", testStream)
				}
			},
			pathParams,
			contentType);

		var apiResponse = new ApiResponse<ApiResponse>(
			(int) response.StatusCode,
			response.Headers.ToDictionary(x => x.Name, x => x.Value.ToString()),
			(ApiResponse)
			client.Deserialize(response, typeof(ApiResponse)));

		Assert.AreEqual(apiResponse.StatusCode, (int) HttpStatusCode.OK);
		Assert.IsTrue(apiResponse.Data.Message.Contains("File uploaded to ./no_file_name_provided, 18 bytes"));
	}
}
```

Reply:
@alex-fisher we do have Petstore integration tests for uploading files to Petstore servers and other operations but seems like they (partially) were removed somehow. I'll need to double check...

Reply:
Is there any further investigation I can do on my part for this request?

Reply:
Let me add back the tests shortly. Stay tuned.

Reply:
Tests added back https://github.com/swagger-api/swagger-codegen/pull/6663

Please rebase on the latest master to add the tests to this PR.

Reply:
thanks for taking that down. I've had a rough week and not much time after work.

Reply:
No worry :)